File: Myecoprint_admin/ncej_admin/lib/data/module/store.dart

class Store {
 int? id;
  String nameArabic;
  String nameEnglish;
  int companyId;
  int typeId;
  String? image;


  Store({
     this.id,
    required this.nameArabic,
    required this.nameEnglish,
    required this.companyId,
    required this.typeId,
     this.image,
 
  });

  factory Store.fromJson(Map<String, dynamic> json) {
    return Store(
     id: json['store_id'] as int? ?? 0,
      nameArabic: json['name_arabic']as String? ?? '',
      nameEnglish: json['name_english']as String? ?? '',
      companyId: json['company_id']as int? ??0,
      typeId: json['type_id']as int? ??0,
      image: json['image']as String? ?? '',

    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id.toString(),
      'name_arabic': nameArabic.toString(),
      'name_english': nameEnglish.toString(),
      'company_id': companyId.toString(),
      'type_id': typeId.toString(),
      'image': image.toString(),
     
    };
  }
}

File: backend_Myecoprint/backend_myecoprint/models/store.js

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const Company = require('./company');
const Type = require('./type');

const Store = sequelize.define('Store', {
    store_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  },
  name_arabic: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  name_english: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  company_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  type_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },  
  image: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  createdAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  updatedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
  deletedAt: {
    type: DataTypes.DATE,
    allowNull: true,
  },
}, {
    timestamps: false, 
    paranoid: true, 
    tableName: 'stores', 
  });
  Store.belongsTo(Company, { foreignKey: 'company_id', as: 'companies',onDelete: 'CASCADE' });
  Store.belongsTo(Type, { foreignKey: 'type_id', as: 'types',onDelete: 'CASCADE' });
 // Store.hasMany(Offer, { foreignKey: 'store_id', as: 'offers' });
module.exports = Store;








